File: 10872.py
listfac = [1]*13

def fact(a):
    if a == 0:
        return 1
    if a == 1:
        return 1
    if listfac[a] != 1:
        return listfac[a]
    else:
        listfac[a] = a * fact(a-1)
        return listfac[a]

n = int(input())
print(fact(n))

File: 10872.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

import java.io.IOException;
 
public class Main {
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int input = Integer.parseInt(br.readLine());
		System.out.println(factorial(input));
	}
 
	public static int factorial(int n) {
		if(n == 1 || n == 0) {
	        return 1;
	    }
	    return n * factorial(n-1);
	}
	
}

